lib/trie/trie.go

```diff
@@ -94,17 +94,17 @@ func (t *Trie) prepBranchForMutation(currentBranch *Node,
 // an older trie generation (snapshot) so we deep copy the
 // node and update the generation on the newer copy.
 func updateGeneration(currentNode *Node, trieGeneration uint64,
-	deletedHashes map[common.Hash]struct{}, copySettings node.CopySettings) (
+	deletedMerkleValues map[string]struct{}, copySettings node.CopySettings) (
 	newNode *Node) {
 	newNode = currentNode.Copy(copySettings)
 	newNode.Generation = trieGeneration
 
 	// The hash of the node from a previous snapshotted trie
 	// is usually already computed.
-	deletedHashBytes := currentNode.MerkleValue
-	if len(deletedHashBytes) > 0 {
-		deletedHash := common.BytesToHash(deletedHashBytes)
-		deletedHashes[deletedHash] = struct{}{}
+	deletedMerkleValueBytes := currentNode.MerkleValue
+	if len(deletedMerkleValueBytes) > 0 {
+		deletedMerkleValueString := string(deletedMerkleValueBytes)
```

Comment:
nit deletedMerkleValue instead, string feels redundant

Reply:
Good point, renamed to `deletedMerkleValue` as suggested